test/grouping.jl

```diff
@@ -1256,4 +1276,131 @@ end
         "│ 2   │ 2     │ 1     │ 2     │\n│ 3   │ 2     │ 2     │ 3     │"
 end
 
+@testset "GroupedDataFrame dictionary interface" begin
+    df = DataFrame(a = repeat([:A, :B, missing], outer=4), b = repeat([:X, :Y], inner=6), c = 1:12)
+    gd = groupby_checked(df, [:a, :b])
+
+    @test map(NamedTuple, keys(gd)) ≅
+        [(a=:A, b=:X), (a=:B, b=:X), (a=missing, b=:X), (a=:A, b=:Y), (a=:B, b=:Y), (a=missing, b=:Y)]
+
+    @test collect(pairs(gd)) ≅ map(Pair, keys(gd), gd)
+
+    for (i, key) in enumerate(keys(gd))
+        # Plain key
+        @test gd[key] ≅ gd[i]
+        # Named tuple
+        @test gd[NamedTuple(key)] ≅ gd[i]
+        # Plain tuple
+        @test gd[Tuple(key)] ≅ gd[i]
+    end
+
+    @test get(gd, (a=:A, b=:X), nothing) ≅ gd[1]
+    @test get(gd, (a=:A, b=:Z), nothing) == nothing
+
+    @test_throws KeyError gd[(a=:A, b=:Z)]
+    @test_throws KeyError gd[(a=:A, b="Z")]
+    @test_throws KeyError gd[(a=:A,)]
+    @test_throws KeyError gd[(a=:A, b=:X, c=1)]
+    @test_throws KeyError gd[(:A, :Z)]
+    @test_throws KeyError gd[(:A,)]
+    @test_throws KeyError gd[(:A, :X, 1)]
+    @test_throws KeyError gd[(b=:X, a=:D)]
+end
+
+@testset "GroupKey and GroupKeys" begin
```

Comment:
do we have tests for invalid keys?

Reply:
What do you mean by an invalid key? I test indexing with `Tuple`s/`NamedTuple`s with incorrect values, incorrect/out of order names, and too many or too few elements, all of which throw a `KeyError`. I also test for using a `GroupKey` from a `GroupedDataFrame` which is equal to but not the same object as the one being indexed. Anything else is a `MethodError`.

Reply:
I meant testing indexing of `GroupKeys` as this is also mentioned in the `testset` name (i.e. that it is only allowed to index them within bounds). This is a trivial test, but it is good to have it (maybe we have it but I missed it), because in the future if someone changes the implementation such trivial things can break.

Reply:
Oh, you mean with integer indices? Yes, that should be added.

Reply:
Yes - thank you!